src/matrix.ts

```diff
@@ -37,7 +37,8 @@ export * from "./models/event-timeline-set";
 export * from "./models/poll";
 export * from "./models/room-member";
 export * from "./models/room-state";
-export { ThreadEvent, ThreadEmittedEvents, ThreadEventHandlerMap, Thread } from "./models/thread";
+export { Thread } from "./models/thread";
+export type { ThreadEvent, ThreadEmittedEvents, ThreadEventHandlerMap } from "./models/thread";
```

Comment:
```suggestion
export * from "./models/thread";
```

The above seem to solve the warning and follows the pattern of what other exports do. I'd suggest moving to that so we can resolve the warning.